src/callstack.cpp

```diff
@@ -82,9 +171,21 @@ static const char* alloc_string( callstack_string_buffer_t* buf, const char* str
 	#   error "Unhandled platform"
 	#endif
 
+	#if defined(__linux)
+		mmap_entry_t *list = parse_mmaps("/proc/self/maps");
```

Comment:
It is a small thing, but maybe it is worth just allocating x mmap_entry_t on the stack. I'm not really sure how many entries might show up in /proc/self/maps? But looking at my machine it is not "that much".
That being said, I'll accept this as well and maybe fix it up later if I need to.

Reply:
Well, something on my system (firefox) has 2000+ lines... More interesting to me would be python, which looks like it starts pretty small (70), but as you import more things, it grows.

Alternatively, if we only care about symbols from the main program (and not any libraries), which is an okay assumption, (we only run addr2line for the main program), then we only need to read the first "r-xp" line from /proc/self/maps.

Reply:
I think this is fine for now. If it turns out to be a problem that can be fixed later.